app/Services/Manager.php

```diff
@@ -80,4 +97,62 @@ public function collectContestInfo($data)
 
         return $contest;
     }
+
+    /**
+     * Get user signup for a specific campaign and run. If only user id is
+     * provided, the send back all user signups.
+     *
+     * @param  string $id
+     * @param  string $campaign    Campaign ID
+     * @param  string $campaignRun Campaign Run ID
+     *
+     * @return object $signups
+     */
+    public function getUserSignup($id, $campaign = null, $campaignRun = null)
+    {
+        $signups = $this->northstar->getUserSignups($id, $campaign);
+
+        // Only return the sign up record for the run that was specified.
+        if ($campaignRun) {
+            foreach ($signups as $key => $signup) {
+                if ($signup->campaign_run->id === (string) $campaignRun) {
```

Comment:
Wait, isn't `$campaignRun` already coming in as a string?

```
@param  string $campaignRun Campaign Run ID
```

And it's the northstar campaign_run id that needs to be changed to a string? Or is it vice versa? If so, update the docblock too, since it's an `int` coming in.